dev/io.openliberty.org.jboss.resteasy.mprestclient/src/org/jboss/resteasy/microprofile/client/ot/LibertyRestClientBuilderImpl.java

```diff
@@ -849,10 +873,27 @@ private String getSystemProperty(String key, String def) {
     }
 
     private static ClassLoader getClassLoader(Class<?> clazz) {
+        ClassLoader clazzLoader = null;
         if (System.getSecurityManager() == null) {
-            return clazz.getClassLoader();
+            clazzLoader = clazz.getClassLoader();
+        } else {
+            clazzLoader = AccessController.doPrivileged((PrivilegedAction<ClassLoader>) clazz::getClassLoader);
+        }
+
+        // We need make sure the correct classloader is used to load files when MPRestClient 
+        // is provided via user OSGI bundles via user feature.
+        // !isOSGiEnv is the case where it is not an OSGi environment.  Mainly this scenario is the TCK scenario.
+        // clazzLoader instanceof EquinoxClassLoader means it is from a Liberty bundle instead of an application
+        try {
+            if (!isOSGiEnv || clazzLoader == null || clazzLoader instanceof EquinoxClassLoader) {
```

Comment:
Do we need this any longer?  Can we just always using the ProxyClassLoader and not worry about OSGi stuff?